src/main/java/spoon/MavenLauncher.java

```diff
@@ -58,50 +61,50 @@
 		ALL_SOURCE
 	}
 
-	public MavenLauncher(String mavenProject, SOURCE_TYPE sourceType) {
-		this(mavenProject, Paths.get(System.getProperty("user.home"), ".m2", "repository").toString(), sourceType);
-	}
-
 	/**
+	 * MavenLauncher constructor assuming either an environment
+	 * variable M2_HOME, or that mvn command exist in PATH.
 	 *
 	 * @param mavenProject the path to the root of the project
-	 * @param m2RepositoryPath the path to the m2repository
 	 * @param sourceType the source type (App, test, or all)
 	 */
-	public MavenLauncher(String mavenProject, String m2RepositoryPath, SOURCE_TYPE sourceType) {
```

Comment:
Actually instead of breaking the compatibility by removing this constructor, you could just deprecate this method and document the fact that m2repositoryPath is not using anymore. 
Same for the other constructor: it would be removed in the next release +1

Reply:
Yes, that makes sense. I'll do that ASAP.